django/http/request.py

```diff
@@ -347,12 +417,12 @@ def __init__(self, query_string=None, mutable=False, encoding=None):
                 except UnicodeDecodeError:
                     # ... but some user agents are misbehaving :-(
                     query_string = query_string.decode('iso-8859-1')
-            for key, value in parse_qsl(query_string or '',
+            for key, value in parse_qsl(SplitWrapper(query_string or '', limit_checker),
```

Comment:
Your approach have the unfortunate effect that it will break if the implementation of parse_qsl() function changes in the future, so that it does not use str.split() anymore.